File: sdks/akta-docs/java/src/main/java/dev/openakta/aktadocs/ConfigLoader.java
package dev.openakta.aktadocs;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;

import java.io.IOException;
import java.nio.file.Path;

public final class ConfigLoader {

    private static final ObjectMapper YAML = new ObjectMapper(new YAMLFactory());

    private ConfigLoader() {}

    public static AktaConfig load(Path path) throws IOException {
        return YAML.readValue(path.toFile(), AktaConfig.class);
    }
}

File: sdks/akta-docs/csharp/ConfigLoader.cs
using YamlDotNet.Serialization;
using YamlDotNet.Serialization.NamingConventions;

namespace OpenAkta.AktaDocs;

public sealed class ConfigException : IOException
{
    public ConfigException(string message, Exception? inner = null) : base(message, inner) { }
}

public static class ConfigLoader
{
    private static readonly IDeserializer Deserializer =
        new DeserializerBuilder()
            .IgnoreUnmatchedProperties()
            .WithNamingConvention(UnderscoredNamingConvention.Instance)
            .Build();

    public static AktaConfig Load(string path)
    {
        string raw;
        try
        {
            raw = File.ReadAllText(path);
        }
        catch (Exception e)
        {
            throw new ConfigException($"Cannot read config: {path}", e);
        }

        try
        {
            return Deserializer.Deserialize<AktaConfig>(raw)
                   ?? throw new ConfigException($"Invalid .akta-config.yaml: {path}");
        }
        catch (Exception e) when (e is not ConfigException)
        {
            throw new ConfigException($"Invalid YAML in {path}", e);
        }
    }

    public static string ResolveConfigPath(string cwd, string? explicitPath) =>
        Path.GetFullPath(explicitPath ?? Path.Combine(cwd, ".akta-config.yaml"));
}
